scipy/optimize/tests/test_optimize.py

```diff
@@ -932,11 +935,15 @@ def test_respect_maxiter(self, method):
             kwargs['hess'] = sf.hess
 
         sol = optimize.minimize(sf.fun, x0, **kwargs)
-        assert sol.nit <= MAXITER
+        assert sol.nit == MAXITER
         assert sol.nfev >= sf.nfev
         if hasattr(sol, 'njev'):
             assert sol.njev >= sf.ngev
 
+        # method specific tests
+        if method in ('SLSQP',):
```

Comment:
I assume you're setting up to add more of these. Although this collection has only one element, a set seems more appropriate to look for something in.
```suggestion
        if method in {'SLSQP'}:
```
Otherwise, why not `if method == 'SLSQP'`?

Reply:
if method == 'SLSQP' is fine. I fixed it.